ext/standard/string.c

```diff
@@ -2204,7 +2202,7 @@ PHP_FUNCTION(substr)
 		} else {
 			f = (zend_long)ZSTR_LEN(str) + f;
 		}
-		if (argc > 2) {
+		if (argc > 2 && !len_is_null) {
```

Comment:
The argc check here should no longer be necessary, just `!len_is_null` should be enough.

Reply:
Good catch! The argc check (and now-unused variable) have been removed.